components/omega/doc/userGuide/OmegaBuild.md

```diff
@@ -0,0 +1,130 @@
+(omega-user-build)=
+
+Omega's build system is built upon the CMake build tool, which provides
+a robust foundation for managing the build process.
+
+The Omega build system supports two modes: standalone and E3SM component.
+
+## Standalone Build
+
+CMake and OMEGA prefer an out-of-source build. This enables a user to build
+and maintain multiple executables from the same source directory.
+The standard practice is for a user to create a separate directory where
+the build should take place and the commands below should be launched from
+that directory.
+
+To perform a standalone build, you need to execute the cmake command with
+the required CMake and Omega parameters. For example, you can specify the
+CC parameter as the C++ compiler and enable the ctest option to guide the
+Omega build process. The following example demonstrates building
+a standalone Omega with a test suite.
+
+```sh
+>> cmake \
+  -DOMEGA_BUILD_TEST=ON \
+  -DOMEGA_CXX_COMPILER=CC \
+  ${E3SM_HOME}/components/omega
+```
+
+Once the `cmake` command succeeds, the directory where the command was
+executed will contain the following files and directories.
+
+```sh
+>> ls -1 ${WORKDIR}
+CMakeCache.txt
+CMakeFiles
+cmake_install.cmake
+CTestTestfile.cmake
+external
+Makefile
+src
+test
+```
+
+To build the Omega, execute the `make` command in the directory.
+
+```sh
+>> make
+Scanning dependencies of target yakl
+[ 11%] Building Fortran object external/YAKL/CMakeFiles/yakl.dir/src/YAKL_gator_mod.F90.o
+[ 22%] Building CXX object external/YAKL/CMakeFiles/yakl.dir/src/YAKL.cpp.o
+[ 33%] Linking CXX static library libyakl.a
+[ 33%] Built target yakl
+[ 44%] Building CXX object src/CMakeFiles/OmegaLib.dir/ocn/ocndummy.cpp.o
+[ 55%] Linking CXX static library libOmegaLib.a
+[ 55%] Built target OmegaLib
+[ 66%] Building CXX object src/CMakeFiles/omega.exe.dir/drivers/drvdummy.cpp.o
+[ 77%] Linking CXX executable omega.exe
+[ 77%] Built target omega.exe
+[ 88%] Building CXX object test/CMakeFiles/OMEGA_TEST.dir/testdummy.cpp.o
+[100%] Linking CXX executable OMEGA_TEST
+[100%] Built target OMEGA_TEST
+```
+
+To run the Omega test suite, execute the `ctest` command.
+
+```sh
+>> ctest
+Test project <cmake working directory>
+    Start 1: OMEGA_TEST
+1/1 Test #1: OMEGA_TEST .......................   Passed    0.01 sec
+
+100% tests passed, 0 tests failed out of 1
+
+Total Test time (real) =   0.01 sec
+```
+
```

Comment:
Ok, one last bit here for standalone builds. We should probably have text here to point to where the omega.exe ends up after a build. For both cases with and without OMEGA_INSTALL_PREFIX. And add the "make install" step for the latter.

Reply:
I updated the user guide with the explanation of the output paths and the usage of OMEGA_INSTALL_PREFIX.